pkg/image/registryclient/client.go

```diff
@@ -35,6 +35,34 @@ type RepositoryRetriever interface {
 	Repository(ctx context.Context, registry *url.URL, repoName string, insecure bool) (distribution.Repository, error)
 }
 
+// ImageSources is a collection of sources for the image. It is designed as an
+// iterator so that it can discover additional sources as needed.
+type ImageSources interface {
+	// Next returns an image reference string for an image. If there are no more
+	// known sources, it returns an empty string. Whenever it's possible, full
+	// references (i.e. with registry names and tags/digests) should be
+	// preferred.
+	Next() (string, error)
+}
+
+// ImageResolver retrieves image sources for image references.
+type ImageResolver interface {
+	Resolve(image string) (ImageSources, error)
+}
+
+type defaultImageResolver struct{}
```

Comment:
There's no value in having this `defaultImageResolver` which is no-op. `ImageResolver` is an optional element of client. You use it if it was provided.

Reply:
I also don't see a value of the extra ImageSources interface. I have a single ImageResolver interface with ResolveNext(string) (string, error)